### ISSUE TITLE
Dressing does not respect multi line prompts

### ISSUE THREAD

**Issue by miversen33** [stevearc/dressing.nvim#89]
**Describe the bug**
When using dressing in place of `vim.ui.input`, if you have your "prompt" object be multiple lines, dressing forces everything onto one line and truncates its length.

**System information**

- OS: `Linux`
- Neovim version: `NVIM v0.9.0-dev-506+g090048bec`
- Is this related to a specific `vim.ui.select` backend? If so, which one? `I don't think so?`
- Dressing config:

```lua
-- Paste your call to require("dressing").setup(...) in here
require("dressing").setup()
```

**To Reproduce**

1. See below min file

If possible, provide a minimal file that will trigger the issue (see
[tests/manual](https://github.com/stevearc/dressing.nvim/tree/master/tests/manual)
for examples of short ways to call `vim.ui.*`):

```lua
vim.ui.input({
    prompt = "Do the thing\n 1) Yes\n 2) Maybe\n 3) No\n\nPlease select all that apply",
    default = "*"
}, function(choice)
        print(choice)
    end
)
```

**Screenshots**
Screenshot without Dressing
![Screenshot without Dressing](https://user-images.githubusercontent.com/2640668/221417363-8affd783-9763-485c-b79d-6c653c1e6427.png)

Screenshot with Dressing
![Screenshot with Dressing](https://user-images.githubusercontent.com/2640668/221417435-460cb9ff-7254-44a2-8f65-d042a4778f1c.png)

**Additional context**
While this maybe addressable with configuration, IMO this should not be something that needs to be configured to rectify. Dressing is (usually) a plugin that is used by other plugins, not by end users. Which IMO means that it would be unfair to expect users to fix this issue via configuring a dependent plugin of a plugin they are using. 

I do note that `:h dressing-format` seems to call out a way to correct this, but that is then reliant on plugin authors to properly configure dressing to properly handle newlines in an "expected" format. 

I suppose the long short, I would expect that newlines be handled as newlines when added to the title of either the `select` or `input` functions, as they are handled in an expected manner when using standard neovim ui elements.

**Comment by stevearc** [comment]
Hmmm...this is interesting because it is technically something that is supported by the default `vim.ui.input` function, but empirically not many people need multi-line prompts for `vim.ui.input`. Even the use case you outline is more suited for something like the possibly-upcoming [vim.ui.confirm](https://github.com/neovim/neovim/pull/21681) (though for multi-select, that would require the still-in-discussion [vim.ui.select_many](https://github.com/neovim/neovim/pull/18161)).

To display this properly, we would have to have a significantly different UI. The current implementation relies on being able to put the entire prompt into the title on a single line. For this to work properly, we would probably have to create two floating windows: one to display the multiline prompt, and one below it for the actual input.

Given the relative rarity of this and my lack of available time, I'm going to sit on this. I'll review PRs, but can't make any promises about working on this myself.

**Comment by emmanueltouzery** [comment]
so i've just had a problem related to that... i had in my clipboard: `"\ntext`, so the first character is a newline. I got that by pasting something from discord.

I pasted that in the dressing input and the input showed "text" as if it was a single line. However when the callback is called, it receives `""`, so only the first line.

See:
https://github.com/stevearc/dressing.nvim/blob/master/lua/dressing/input.lua#L141

I think dressing should make some effort to return the line that it's displaying, which apparently may not necessarily be the first one.

Note that I could reproduce by selecting character-wise the text with the newline from neovim (v and move the cursor). If I copied line-wise (shift-v) then dressing did display an empty line after pasting.
